lib/observe.js

```diff
@@ -0,0 +1,74 @@
+var _ = require('lodash');
+var merge = require('viewdb').merge;
+var MongoOplog = require('mongo-oplog');
+
+var logger = {
+  added: function() { console.log("added: "); console.log(arguments); },
+  removed: function() { console.log("removed: "); console.log(arguments); },
+  changed: function() { console.log("changed: "); console.log(arguments); },
+  moved: function() { console.log("moved: "); console.log(arguments); },
+};
+
+var oplogUri = 'mongodb://127.0.0.1:27017/local';
+
+var Observer = function (query, queryOptions, collection, options) {
+  var self = this;
+  const namespace = collection._collection.s.namespace;
+  this._query = query;
+  this._options = options;
+  // this._options = logger;
+  this._collection = collection;
+  this._cache = [];
+
+  this._oplog = MongoOplog(oplogUri, { ns: namespace });
+
+  self._oplog.tail().then(function () {
+    self.loadInitial(function () {
+      self._oplog.on('insert', self._onInsertUpdate.bind(self));
+      self._oplog.on('update', self._onInsertUpdate.bind(self));
+      self._oplog.on('delete', self._onRemove.bind(self));
+    });
+  });
+
+  return {
+    stop: function () {
+      self._cache = null;
+      self._oplog.stop();
+      self._oplog.destroy();
+    }
+  }
+};
+
+var comparator = function (a, b) {
+  return a._id === b._id
+};
+
+Observer.prototype.loadInitial = function (cb) {
+  var self = this;
+  this._collection._getDocuments(this._query, function (err, result) {
+    if (self._options.init) {
+      self._cache = result;
+      self._options.init(result);
+    } else {
+      var old = self._cache;
+      self._cache = merge(old, result, _.defaults({ comparatorId: comparator }, self._options));
```

Comment:
tror man egentligen skall skippa cache och merge. bara har en cache av id's och inte hela objektet sedan se om id fanns tidigare isåfall emitta change annars add eller vid remove om det inte fanns tidigare så ignorera etc. men kan optimera det senare